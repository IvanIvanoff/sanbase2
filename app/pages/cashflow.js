import fetch from 'isomorphic-unfetch'
import { WEBSITE_URL } from '../config'
import ProjectsTable from '../components/projects-table'
import MainHead from '../components/main-head'
import SideMenu from '../components/side-menu'

const Index = (props) => (
  <div>
    <MainHead>
      <link rel="stylesheet" href="https://cdn.datatables.net/responsive/2.1.1/css/responsive.dataTables.min.css"/>
      <link rel="stylesheet" href="https://cdn.datatables.net/fixedheader/3.1.2/css/fixedHeader.bootstrap.min.css"/>
      <script src="https://cdn.datatables.net/responsive/2.1.1/js/dataTables.responsive.min.js"></script>
      <script src="https://cdn.datatables.net/responsive/2.1.1/js/responsive.bootstrap.min.js"></script>
      <script src="https://cdn.datatables.net/fixedheader/3.1.2/js/dataTables.fixedHeader.min.js"></script>
    </MainHead>
    <SideMenu activeItem="cashflow"/>
    <div className="container" id="main">
      <div className="row">
          <div className="col-lg-5">
              <h1>Cash Flow</h1>
          </div>
          <div className="col-lg-7 community-actions">
              <span className="legal">brought to you by <a href="https://santiment.net" target="_blank">Santiment</a>
              <br />
              NOTE: This app is a prototype. We give no guarantee data is correct as we are in active development.</span>
          </div>
      </div>
      <ProjectsTable data={ props.data }/>
    </div>
    <script dangerouslySetInnerHTML={{ __html: `
      $(document).ready(function () {

          $('.table-hover').DataTable({

              "dom": "<'row'<'col-sm-7'i><'col-sm-5'f>>" +
              "<'row'<'col-sm-12'tr>>" +
              "<'row'<'col-sm-5'><'col-sm-7'>>",
              "paging": false,
              fixedHeader: true,
              language: {
                  search: "_INPUT_",
                  searchPlaceholder: "Search"
              },
            "order": [[ 1, "desc" ]],

              responsive: {
                  details: {
                      display: $.fn.dataTable.Responsive.display.childRowImmediate,
                      type: ''
                  }
              }

          });

      });
    `}} />
  </div>
)

Index.getInitialProps = async function() {
  //const res = await fetch(WEBSITE_URL + '/api/cashflow')
  //const data = await res.json()

  const data = {
    projects: [
      {
        market_cap_usd: 1,
        balance: 45,
        name: 'EOS',
        ticker: 'EOS',
        logo_url: 'eos.png',
        wallets: [
          {
            last_outgoing: null,
            balance: 1,
            tx_out: null
          }
        ]
      }
    ],
    eth_price: 2
  };

  return {
    data: data
  }
}

export default Index
