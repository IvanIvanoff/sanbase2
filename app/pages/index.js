import Router from 'next/router'

const Index = (props) => {}

Index.getInitialProps = ({res}) => {
  if (res) {
    res.writeHead(301, { Location: '/cashflow' })
    res.end()
    res.finished = true
  } else {
    Router.replace('/cashflow')
  }
  return {}
}

export default Index
