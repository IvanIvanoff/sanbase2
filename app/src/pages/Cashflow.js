import React from 'react'
import 'react-table/react-table.css'
import ProjectsTabel from 'pages/Projects/ProjectsTabel'
import withProjectsData from 'pages/Projects/withProjectsData'

export const Cashflow = ({
  Projects,
  onSearch,
  handleSetCategory,
  history,
  match,
  search,
  tableInfo,
  categories,
  allMarketSegments,
  preload,
  user
}) => (
  <ProjectsTabel
    Projects={Projects}
    onSearch={onSearch}
    handleSetCategory={handleSetCategory}
    history={history}
    match={match}
    search={search}
    tableInfo={tableInfo}
    categories={categories}
    allMarketSegments={allMarketSegments}
    preload={preload}
    user={user}
  />
)

export default withProjectsData({type: 'erc20'})(Cashflow)
