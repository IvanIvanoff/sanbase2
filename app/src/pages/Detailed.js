import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import {
  compose,
  lifecycle
} from 'recompose'
import { Merge, FadeIn } from 'animate-components'
import { fadeIn, slideRight } from 'animate-keyframes'
import { Redirect } from 'react-router-dom'
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs'
import gql from 'graphql-tag'
import { graphql } from 'react-apollo'
import { retrieveProjects } from './Cashflow.actions.js'
import ProjectIcon from './../components/ProjectIcon'
import PanelBlock from './../components/PanelBlock'
import GeneralInfoBlock from './../components/GeneralInfoBlock'
import FinancialsBlock from './../components/FinancialsBlock'
import ProjectChartContainer from './../components/ProjectChart/ProjectChartContainer'
import { formatNumber, formatBTC } from '../utils/formatting'
import Panel from './../components/Panel'
import './Detailed.css'

const propTypes = {
  match: PropTypes.object.isRequired,
  projects: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired,
  generalInfo: PropTypes.object
}

export const HiddenElements = () => ''

const getProjectByTicker = (match, projects) => {
  const selectedTicker = match.params.ticker
  const project = projects.find(el => {
    const ticker = el.ticker || ''
    return ticker.toLowerCase() === selectedTicker
  })
  return project
}

export const Detailed = ({
  match,
  projects,
  loading,
  PriceQuery,
  generalInfo
}) => {
  if (loading) {
    return (
      <div className='page detailed'>
        <h2>Loading...</h2>
      </div>
    )
  }
  const project = getProjectByTicker(match, projects)

  if (!project) {
    return (
      <Redirect to={{
        pathname: '/'
      }} />
    )
  }

  return (
    <div className='page detailed'>
      <FadeIn duration='0.7s' timingFunction='ease-in' as='div'>
        <div className='detailed-head'>
          <div className='detailed-name'>
            <h1><ProjectIcon name={project.name} size={28} /> {project.name} ({project.ticker.toUpperCase()})</h1>
            <p>Manage entire organisations using the blockchain.</p>
          </div>

          {!PriceQuery.loading && PriceQuery.price &&
            <Merge
              one={{ name: fadeIn, duration: '0.3s', timingFunction: 'ease-in' }}
              two={{ name: slideRight, duration: '0.5s', timingFunction: 'ease-out' }}
              as='div'
            >
              <div className='detailed-price'>
                <div>{formatNumber(PriceQuery.price.priceUsd, 'USD')}</div>
                <div>BTC {formatBTC(parseFloat(PriceQuery.price.priceBtc))}</div>
              </div>
            </Merge>}

          <HiddenElements>
            <div className='detailed-buttons'>
              <button className='add-to-dashboard'>
                <i className='fa fa-plus' />
                &nbsp; Add to Dashboard
              </button>
            </div>
          </HiddenElements>
        </div>
        <Panel withoutHeader>
          <ProjectChartContainer ticker={project.ticker} />
        </Panel>
        <HiddenElements>
          <div className='panel'>
            <Tabs className='activity-panel'>
              <TabList className='nav'>
                <Tab className='nav-item' selectedClassName='active'>
                  <button className='nav-link'>
                    Social Mentions
                  </button>
                </Tab>
                <Tab className='nav-item' selectedClassName='active'>
                  <button className='nav-link'>
                    Social Activity over Time
                  </button>
                </Tab>
                <Tab className='nav-item' selectedClassName='active'>
                  <button className='nav-link'>
                    Sentiment/Intensity
                  </button>
                </Tab>
                <Tab className='nav-item' selectedClassName='active'>
                  <button className='nav-link'>
                    Github Activity
                  </button>
                </Tab>
                <Tab className='nav-item' selectedClassName='active'>
                  <button className='nav-link'>
                    SAN Community
                  </button>
                </Tab>
              </TabList>
              <TabPanel>
                Social Mentions
              </TabPanel>
              <TabPanel>
                Social Activity over Time
              </TabPanel>
              <TabPanel>
                Sentiment/Intensity
              </TabPanel>
              <TabPanel>
                Github Activity
              </TabPanel>
              <TabPanel>
                SAN Community
              </TabPanel>
            </Tabs>
          </div>
        </HiddenElements>
        <HiddenElements>
          <PanelBlock title='Blockchain Analytics' />
          <div className='analysis'>
            <PanelBlock title='Signals/Volatility' />
            <PanelBlock title='Expert Analyses' />
            <PanelBlock title='News/Press' />
          </div>
        </HiddenElements>
        <div className='information'>
          <PanelBlock
            isUnauthorized={generalInfo.isUnauthorized}
            isLoading={generalInfo.isLoading || PriceQuery.loading}
            title='General Info'>
            <GeneralInfoBlock
              volume={!!PriceQuery.price && PriceQuery.price.volume}
              {...generalInfo.project} />
          </PanelBlock>
          <PanelBlock
            isUnauthorized={generalInfo.isUnauthorized}
            isLoading={generalInfo.isLoading}
            title='Financials'>
            <FinancialsBlock {...generalInfo.project} />
          </PanelBlock>
        </div>
      </FadeIn>
    </div>
  )
}

Detailed.propTypes = propTypes

const mapStateToProps = state => {
  return {
    projects: state.projects.items,
    loading: state.projects.isLoading
  }
}

const mapDispatchToProps = dispatch => {
  return {
    // TODO: have to retrive only selected project
    retrieveProjects: () => dispatch(retrieveProjects)
  }
}

const getPriceGQL = gql`
  query getPrice($ticker: String!) {
    price (
      ticker: $ticker
    ) {
      priceBtc,
      priceUsd,
      volume,
      datetime,
      marketcap
    }
}`

const queryProject = gql`
  query project($id: ID!) {
    project(
      id: $id,
    ){
      id,
      name,
      ticker,
      marketCapUsd,
      websiteLink,
      facebookLink,
      githubLink,
      redditLink,
      twitterLink,
      whitepaperLink,
      slackLink,
      btcBalance,
      projectTransparency,
      projectTransparencyDescription,
      projectTransparencyStatus,
      tokenAddress,
      fundsRaisedIcos { amount, currencyCode },
      latestCoinmarketcapData {
        priceUsd,
        updateTime,
        marketCapUsd
      }
    }
  }
`

const mapDataToProps = ({ProjectQuery}) => {
  const isLoading = ProjectQuery.loading
  const isEmpty = !!ProjectQuery.project
  const isError = !!ProjectQuery.error
  const errorMessage = ProjectQuery.error ? ProjectQuery.error.message : ''
  const isUnauthorized = ProjectQuery.error ? /\bunauthorized/.test(ProjectQuery.error.message) : false
  if (ProjectQuery.error && !isUnauthorized) {
    // If our API server is not reponed with ProjectQuery
    throw new Error(ProjectQuery.error.message)
  }
  const project = ProjectQuery.project

  return {generalInfo: {isLoading, isEmpty, isError, project, errorMessage, isUnauthorized}}
}

const mapPropsToOptions = ({match, projects}) => {
  const project = getProjectByTicker(match, projects)
  return {
    skip: !project,
    variables: {
      id: project ? project.id : 0
    }
  }
}

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  graphql(getPriceGQL, {
    name: 'PriceQuery',
    options: ({match, projects}) => {
      const project = getProjectByTicker(match, projects)
      return {
        skip: !project,
        variables: {
          'ticker': project ? project.ticker.toUpperCase() : 'SAN'
        }
      }
    }
  }),
  graphql(queryProject, {
    name: 'ProjectQuery',
    props: mapDataToProps,
    options: mapPropsToOptions
  }),
  lifecycle({
    componentDidMount () {
      this.props.retrieveProjects()
    }
  })
)

export default enhance(Detailed)
