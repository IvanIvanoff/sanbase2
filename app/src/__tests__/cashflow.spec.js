/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Cashflow } from './../pages/Cashflow'

describe('Cashflow container', () => {
  it('it should render correctly', () => {
    const login = shallow(<Cashflow />)
    expect(toJson(login)).toMatchSnapshot()
  })
})
