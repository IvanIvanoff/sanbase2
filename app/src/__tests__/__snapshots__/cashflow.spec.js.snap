// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cashflow container it should render correctly 1`] = `
<div
  className="page cashflow"
>
  <div
    className="cashflow-head"
  >
    <h1>
      Cash Flow
    </h1>
    <p>
      brought to you by 
      <a
        href="https://santiment.net"
        rel="noopener noreferrer"
        target="_blank"
      >
        Santiment
      </a>
      <br />
      NOTE: This app is a prototype. We give no guarantee data is correct as we are in active development.
    </p>
  </div>
  <div
    className="panel"
  >
    <div
      className="row"
    >
      <div
        className="datatables-info"
      >
        <label>
          Showing 
          1
           to 
          32
           of 
          32
           entries 
          (filtered from 2 total entries)
        </label>
      </div>
      <div
        className="datatables-filter"
      >
        <label>
          <input
            placeholder="Search"
          />
        </label>
      </div>
    </div>
    <ReactTable
      AggregatedComponent={[Function]}
      ExpanderComponent={[Function]}
      FilterComponent={[Function]}
      LoadingComponent={[Function]}
      NoDataComponent={[Function]}
      PadRowComponent={[Function]}
      PaginationComponent={[Function]}
      PivotValueComponent={[Function]}
      ResizerComponent={[Function]}
      TableComponent={[Function]}
      TbodyComponent={[Function]}
      TdComponent={[Function]}
      TfootComponent={[Function]}
      ThComponent={[Function]}
      TheadComponent={[Function]}
      TrComponent={[Function]}
      TrGroupComponent={[Function]}
      aggregatedKey="_aggregated"
      className="-highlight"
      collapseOnDataChange={true}
      collapseOnPageChange={true}
      collapseOnSortingChange={true}
      column={
        Object {
          "Aggregated": undefined,
          "Cell": undefined,
          "Expander": undefined,
          "Filter": undefined,
          "Footer": undefined,
          "Header": undefined,
          "Pivot": undefined,
          "PivotValue": undefined,
          "aggregate": undefined,
          "className": "",
          "filterAll": false,
          "filterMethod": undefined,
          "filterable": undefined,
          "footerClassName": "",
          "footerStyle": Object {},
          "getFooterProps": [Function],
          "getHeaderProps": [Function],
          "getProps": [Function],
          "headerClassName": "",
          "headerStyle": Object {},
          "minWidth": 100,
          "resizable": undefined,
          "show": true,
          "sortMethod": undefined,
          "sortable": undefined,
          "style": Object {},
        }
      }
      columns={
        Array [
          Object {
            "Cell": [Function],
            "Header": "Project",
            "accessor": [Function],
            "filterMethod": [Function],
            "filterable": true,
            "id": "project",
            "sortable": true,
            "width": 350,
          },
          Object {
            "Cell": [Function],
            "Header": "Market Cap",
            "accessor": "market_cap_usd",
            "id": "market_cap_usd",
            "minWidth": 150,
            "sortMethod": [Function],
            "sortable": true,
          },
          Object {
            "Cell": [Function],
            "Header": "Balance (USD/ETH)",
            "accessor": [Function],
            "id": "balance",
            "minWidth": 250,
            "sortMethod": [Function],
            "sortable": true,
          },
          Object {
            "Cell": [Function],
            "Header": "Last outgoing TX",
            "accessor": [Function],
            "id": "tx",
            "sortMethod": [Function],
            "sortable": true,
          },
          Object {
            "Cell": [Function],
            "Header": "ETH sent",
            "accessor": [Function],
            "id": "sent",
            "sortMethod": [Function],
            "sortable": true,
          },
        ]
      }
      data={
        Array [
          Object {
            "balance": "41994.09",
            "id": 12,
            "logo_url": "aeternity.png",
            "market_cap_usd": "133759577.0",
            "name": "Aeternity",
            "ticker": "AE",
            "wallets": Array [
              Object {
                "address": "0x15c19E6c203E2c34D1EDFb26626bfc4F65eF96F0",
                "balance": "41994.09",
                "last_outgoing": null,
                "tx_out": null,
              },
            ],
          },
          Object {
            "balance": "264617.29",
            "id": 15,
            "logo_url": "aragon.png",
            "market_cap_usd": "66649211.0",
            "name": "Aragon",
            "ticker": "ANT",
            "wallets": Array [
              Object {
                "address": "0xcafE1A77e84698c83CA8931F54A755176eF75f2C",
                "balance": "264617.29",
                "last_outgoing": null,
                "tx_out": null,
              },
            ],
          },
        ]
      }
      defaultExpanded={Object {}}
      defaultFilterMethod={[Function]}
      defaultFiltered={Array []}
      defaultPageSize={32}
      defaultResized={Array []}
      defaultSortDesc={false}
      defaultSortMethod={[Function]}
      defaultSorted={
        Array [
          Object {
            "desc": false,
            "id": "market_cap_usd",
          },
        ]
      }
      expanderDefaults={
        Object {
          "filterable": false,
          "resizable": false,
          "sortable": false,
          "width": 35,
        }
      }
      filterable={false}
      filtered={Array []}
      freezeWhenExpanded={false}
      getLoadingProps={[Function]}
      getNoDataProps={[Function]}
      getPaginationProps={[Function]}
      getProps={[Function]}
      getResizerProps={[Function]}
      getTableProps={[Function]}
      getTbodyProps={[Function]}
      getTdProps={[Function]}
      getTfootProps={[Function]}
      getTfootTdProps={[Function]}
      getTfootTrProps={[Function]}
      getTheadFilterProps={[Function]}
      getTheadFilterThProps={[Function]}
      getTheadFilterTrProps={[Function]}
      getTheadGroupProps={[Function]}
      getTheadGroupThProps={[Function]}
      getTheadGroupTrProps={[Function]}
      getTheadProps={[Function]}
      getTheadThProps={[Function]}
      getTheadTrProps={[Function]}
      getTrGroupProps={[Function]}
      getTrProps={[Function]}
      groupedByPivotKey="_groupedByPivot"
      indexKey="_index"
      loading={false}
      loadingText="Loading..."
      nestingLevelKey="_nestingLevel"
      nextText="Next"
      noDataText="No rows found"
      ofText="of"
      onFetchData={[Function]}
      originalKey="_original"
      pageSizeOptions={
        Array [
          5,
          10,
          20,
          25,
          50,
          100,
        ]
      }
      pageText="Page"
      pivotDefaults={Object {}}
      pivotIDKey="_pivotID"
      pivotValKey="_pivotVal"
      previousText="Previous"
      resizable={true}
      rowsText="rows"
      showPageJump={true}
      showPageSizeOptions={true}
      showPagination={false}
      showPaginationBottom={false}
      showPaginationTop={false}
      sortable={false}
      style={Object {}}
      subRowsKey="_subRows"
    />
  </div>
</div>
`;
